okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -414,9 +445,12 @@ public void shutdown() {
       }
     }
 
-    // Send GOAWAY with lastGoodStreamId of 0, since we don't expect any server-initiated streams.
-    // The GOAWAY is part of graceful shutdown.
-    frameWriter.goAway(0, ErrorCode.NO_ERROR, new byte[0]);
+    // If the transport was never started, shutdown should not cause an error.
```

Comment:
@madongfly, how do you feel about this? It is fine with me, but in some ways I think it would be more reliable and maintainable to prohibit calling shutdown if start wasn't called. I can't think of a great case that would benefit from this, where this solution is better than the alternatives.